tests/foreman/ui_airgun/test_discoveryrule.py

```diff
@@ -295,3 +295,85 @@ def test_positive_list_host_based_on_rule_search_query(
         values = session.host.get_details(provisioned_host_name)
         assert (values['properties']['properties_table']['IP Address']
                 == ip_address)
+
+
+@tier2
+@upgrade
+def test_positive_end_to_end(session, module_org, module_loc):
+    """Perform end to end testing for discovery rule component.
+
+    :id: dd35e566-dc3a-43d3-939c-a33ae528740f
+
+    :expectedresults: All expected CRUD actions finished successfully
+
+    :CaseLevel: Integration
+
+    :CaseImportance: High
+    """
+
+    rule_name = gen_string('alpha')
+    search = f'cpu_count = {gen_integer(1, 5)}'
+    hg_name = gen_string('alpha')
+    hostname = gen_string('alpha')
+    hosts_limit = str(gen_integer(0, 100))
+    priority = str(gen_integer(1, 100))
+    new_rule_name = gen_string('alpha')
+    new_search = f'cpu_count = {gen_integer(6, 10)}'
+    new_hg_name = gen_string('alpha')
+    new_hostname = gen_string('alpha')
+    new_hosts_limit = str(gen_integer(101, 200))
+    new_priority = str(gen_integer(101, 200))
+    entities.HostGroup(name=hg_name, organization=[module_org], location=[module_loc]).create()
+    entities.HostGroup(name=new_hg_name, organization=[module_org], location=[module_loc]).create()
+    new_org = entities.Organization().create()
+    new_loc = entities.Location().create()
+    with session:
+        session.discoveryrule.create({
+            'primary.name': rule_name,
+            'primary.search': search,
+            'primary.host_group': hg_name,
```

Comment:
can we check that rule is assigned to host group from host group side?

Reply:
Unfortunately we can not have that value from host group side in edit view nor in host group search.